Mathlib/GroupTheory/GroupAction/Period.lean

```diff
@@ -0,0 +1,113 @@
+/-
+Copyright (c) 2024 Emilie Burgun. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Emilie Burgun
+-/
+
+import Mathlib.GroupTheory.GroupAction.Basic
+import Mathlib.Dynamics.PeriodicPts
+import Mathlib.GroupTheory.Exponent
+
+/-!
+# Period of a group action
+
+This module defines some helpful lemmas around [`MulAction.period`] and [`AddAction.period`].
+The period of a point `a` by a group element `g` is the smallest `m` such that `g ^ m • a = a`
+(resp. `(m • g) +ᵥ a = a`) for a given `g : G` and `a : α`.
+
+If such an `m` does not exist,
+then by convention `MulAction.period` and `AddAction.period` return 0.
+-/
+
+namespace MulAction
+
+universe u v
+variable {α : Type v}
+variable {G : Type u} [Group G] [MulAction G α]
+variable {M : Type u} [Monoid M] [MulAction M α]
+
+/-- If the action is periodic, then a lower bound for its period can be computed. -/
+@[to_additive]
+theorem period_gt_of_moved {m : M} {a : α} {n : ℕ} (period_pos : 0 < period m a)
```

Comment:
Theorem names really have to closely match their statement, so:
```suggestion
theorem le_period_of_moved {m : M} {a : α} {n : ℕ} (period_pos : 0 < period m a)
```

(I assume "moved" is used as the opposite of "fixed (point)"? I'm not familiar with any word for this concept, so maybe another name would be clearer for other users too.)

Reply:
To be fair, this is the only real mechanism to prove a lower bound for the period, so we could just go with `le_period`